src/screens/AddCash/AddCash.js

```diff
@@ -0,0 +1,138 @@
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import React from 'react';
+import { ScrollView, Keyboard } from 'react-native';
+import t from 'translations/translate';
+import styled from 'styled-components/native';
+import { useNavigation } from 'react-navigation-hooks';
+
+
+// utils
+import { fontSizes, appFont } from 'utils/variables';
+import { isValidFiatValue } from 'utils/validators';
+import { getCurrencySymbol } from 'utils/common';
+import { openInAppBrowser } from 'utils/inAppBrowser';
+import { rampWidgetUrl } from 'utils/fiatToCrypto';
+
+// compomnents
+import { Container } from 'components/modern/Layout';
+import { Footer } from 'components/Layout/Layout';
+import Button from 'components/Button';
+import TextInput from 'components/TextInput';
+import HeaderBlock from 'components/HeaderBlock';
+import Text from 'components/modern/Text';
+import Toast from 'components/Toast';
+
+// selectors
+import { useFiatCurrency } from 'selectors';
+
+
+const AddCash = () => {
+  const navigation = useNavigation();
+  const [value, setValue] = React.useState('0');
+  const fiatCurrency = useFiatCurrency();
+  const currencySymbol = getCurrencySymbol(fiatCurrency);
+  const address = navigation.getParam('address');
+  const email = navigation.getParam('email');
+
+  const tryOpenCryptoPurchaseUrl = async (url: string | null) => {
+    if (url) {
+      await openInAppBrowser(url).catch(showServiceLaunchError);
+    } else {
+      showServiceLaunchError();
+    }
+  };
+
+  const showServiceLaunchError = () => {
+    Toast.show({
+      message: t('toast.cryptoPurchaseLaunchFailed'),
+      emoji: 'hushed',
+      supportLink: true,
+    });
+  };
+
+  return (
+    <Container>
+      <HeaderBlock
+        centerItems={[{ title: t('servicesContent.ramp.addCash.title') }]}
+        leftItems={[{ close: true }]}
+        navigation={navigation}
+      />
+      <ScrollView onScroll={() => Keyboard.dismiss()} keyboardShouldPersistTaps="handled">
+        <AddCashView>
+          <Text variant="medium">{t('servicesContent.ramp.addCash.subTitle')}</Text>
+          <TextInput
+            inputProps={{
+              value: `${currencySymbol}${value}`,
+              autoCapitalize: 'none',
+              disabled: false,
+              onChangeText: (text) => {
+                const amount = text.replace(currencySymbol, '');
+                const updatedAmount = amount.replace(/^0+/, '');
+                setValue(updatedAmount);
+              },
+              placeholder: '$0',
+              keyboardType: 'numeric',
+            }}
+            inputWrapperStyle={styles.inputWrapperStyles}
+            itemHolderStyle={styles.itemHolderStyles}
+            additionalStyle={styles.additionalStyle}
+            errorMessage={!isValidFiatValue(value) && t('error.invalid.fiatValue')}
```

Comment:
We should also check for value greater than zero, as purchasing for 0 USD makes little sense.